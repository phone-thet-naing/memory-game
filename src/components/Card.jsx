/* eslint-disable react/prop-types */
// import { PropTypes } from 'prop-types';

function Card({ image, selected, onClick }) {
	return (
		<div className='card' onClick={onClick}>
			<div className={selected && 'selected'}>
				<img src={image} alt='' className='card-face' />

				<img src='/src/assets/back.svg' alt='' className='card-back' />
			</div>
		</div>
	);
}

export default Card;
