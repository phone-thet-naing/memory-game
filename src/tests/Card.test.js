import renderer from 'react-test-renderer';
import Card from '../components/Card';

/*global 
it
expect
 */
it('case 1', () => {
	const cardComponent = renderer.create(
		<Card image={'./src/assets/back.svg'} selected={true} />
	);

	let tree = cardComponent.toJSON();
	expect(tree).toMatchSnapshot();

	// manually trigger the callback
	// renderer.act(() => {
	// 	tree.props.onClick();
	// });
	// // re-rendering
	// tree = cardComponent.toJSON();
	// console.log(tree);
	// expect(tree).toMatchSnapshot();
});
