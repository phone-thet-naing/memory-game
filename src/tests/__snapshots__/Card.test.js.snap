// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`case 1 1`] = `
<div
  className="card"
>
  <div
    className="selected"
  >
    <img
      alt=""
      className="card-face"
      src="./src/assets/back.svg"
    />
    <img
      alt=""
      className="card-back"
      src="/src/assets/back.svg"
    />
  </div>
</div>
`;
